### PR TITLE
CICE_CPPDEFS was not being set correctly in some cases.

### PR DESCRIPTION
**Pull request by jgfouca**
Also, some minor fixes to CIME to catch this error sooner.

[BFB]

### CODE REVIEW

File: components/cice/cime_config/buildnml [hunk 1]
```diff
@@ -70,9 +70,10 @@ def buildnml(case, caseroot, compname):
                                  from_dir=os.path.join(srcroot, "components/cice/bld")).split()
 
         if int(config[0]) >= 0:
-            case.set_value("CICE_BLCKX",         config[2])
-            case.set_value("CICE_BLCKY",         config[3])
-            case.set_value("CICE_MXBLCKS",       config[4])
+            cice_blckx, cice_blcky, cice_mxblcks = config[2:5]
```
Comment (jgfouca) [inline components/cice/cime_config/buildnml]: The bug is here. The original code has the potential for case's CICE_BLCKX, CICE_BLCKY, and CICE_MXBLCKS to become out-of-sync with the local variables cice_blckx, etc which are used to define CICE_CPPDEFS below.

**Comment by jgfouca** [comment]
@oksanaguba , once this PR is merged, if you rebase your #3009 PR, it should work with those ERS tests in the same shared test suite.

**Comment by oksanaguba** [comment]
@jgfouca thanks -- I will make a note about this.

**Comment by rljacob** [comment]
You can merge this.

**Pull request merged**